docs/js/Expression.js

```diff
@@ -0,0 +1,59 @@
+import { UnknownOperator } from "./Errors"
+import { InputReader } from "./InputReader"
+import { Operators } from "./Operators"
+import { Brackets } from "./Brackets"
+
+
+export class Expression {
+    constructor(input) {
+        this.#reader = new InputReader(input)
+
+        this.#operands = []
+        this.#operators = []
+    }
+
+    #reader
+    #operands
+    #operators
+
+    solve() {
+        for (const token of this.#reader.iterateTokens()) {
+            if (typeof token === "number") {
+                this.#operands.push(token)
+            } else if (Brackets.isOpen(token)) {
+                this.#operators.push(token)
+            } else if (Brackets.isClosed(token)) {
+                while (!Brackets.isOpen(this.#peekOperators())) {
+                    this.#operateTop()
+                }
+                this.#operators.pop()
+            } else if (Operators.isOperator(token)) {
+                while ((this.#operators.length !== 0) &&
+                (Operators.compare(this.#peekOperators(), token) >= 0)) {
```

Comment:
This is unclean:
1. The condition in `if`, `while`, `for` should ideally fit into one line. If it doesn't it's best to extract it to a method instead of introducing a line break.
2. Even if you introduce a line break, the next line should be indented properly.
3. In this case, you don't need to do all that because the `length` check is redundant. If you delete it, the tests should still pass just fine. If you don't understand why, look into the implementation of `Operators.compare` and read up on the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing).